modules/ding_library/ding_library.module

```diff
@@ -216,7 +216,13 @@ function ding_library_disable_custom_alias($form, &$form_state) {
     $form['path']['pathauto']['#description'] = t('Custom alias has been disabled');
     $form['path']['pathauto']['#attached']['js'][] = drupal_get_path('module', 'ding_library') . '/ding_library.js';
 
-    unset($form['path']['alias']);
+    // We have to keep the alias textfield, since removing it causes a fatal
+    // error in the parent element validate handler, so we just tell the FORM
+    // API it's disabled and make sure it's disabled in frontend.
+    $form['path']['alias']['#disabled'] = TRUE;
+    // For some reason this is necessary.
+    $form['path']['alias']['#attributes'] = array('disabled' => TRUE);
+    unset($form['path']['alias']['#states']);
```

Comment:
Have you tried setting `$form['path']['#access']` to `FALSE` instead of `$form['path']['alias']['#disabled'] = TRUE;`? This way the form element will not appear in the rendered form but it will still be available serverside.

This might also remove the need for modifying `#attributes` and `#states`.